website/frontend/src/components/App/App.tsx

```diff
@@ -54,34 +55,25 @@ const utilRoutes = [routes.privacyPolicy, routes.termsOfUse];
 
 const AppComponent: FC = () => {
   globalStyles();
-  useBlocks();
-  const dispatch = useDispatch();
+  const { isApiReady } = useApi();
+  const alert = useAlert();
+  const { isLoading } = useLoading();
   const [searchParams, setSearchParams] = useSearchParams();
-  const { isApiReady } = useSelector((state: RootState) => state.api);
-  const { isProgramUploading, isMessageSending } = useSelector((state: RootState) => state.programs);
   const events = useEvents();
 
   useEffect(() => {
-    if ((isProgramUploading || isMessageSending) && document.body.style.overflowY !== 'hidden') {
+    if (isLoading && document.body.style.overflowY !== 'hidden') {
```

Comment:
could you please leave here TODO about change this hack to useBodyScrollLock hook?

Reply:
It's obvious that this part should be refactored so no need for additional comment I guess